pkg/codegen/dotnet/gen_program.go

```diff
@@ -1321,6 +1351,9 @@ func (g *generator) genResourceOptions(opts *pcl.ResourceOptions, resourceOption
 	if opts.HideDiffs != nil {
 		appendOption("HideDiffs", opts.HideDiffs)
 	}
+	if opts.ReplacementTrigger != nil {
```

Comment:
This is the thing I actually came here to fix, but I'll surface this in the dotnet changelog.